packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -97,86 +115,101 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
 
         Commitment storage commitment = lenderMarketCommitments[lender][
             _marketId
-        ][_tokenAddress];
+        ][_principalTokenAddress];
```

Comment:
lets also add `_collateralTokenAddress` as an additional nested `mapping` to this storage variable
i.e.
```solidity
        Commitment storage commitment = lenderMarketCommitments[lender][
            _marketId
        ][_principalTokenAddress][_collateralTokenAddress];
```
this way a lender can have multiple commitments in a market specified for different collateral tokens

NOTE: there is an additional task to be worked on to allow lenders to specific a borrower address as well which would then add yet another `mapping` - perhaps theres a better solution to this that we can discuss